chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc

```diff
@@ -385,11 +385,13 @@ void BraveRenderViewContextMenu::CopyTextFromImage() {
 
 #if BUILDFLAG(ENABLE_AI_CHAT)
 bool BraveRenderViewContextMenu::IsAIChatEnabled() const {
-  return ai_chat::IsAIChatEnabled(GetProfile()->GetPrefs()) &&
+  return !params_.selection_text.empty() &&
+         ai_chat::IsAIChatEnabled(GetProfile()->GetPrefs()) &&
          brave::IsRegularProfile(GetProfile()) &&
          GetProfile()->GetPrefs()->GetBoolean(
              ai_chat::prefs::kBraveAIChatContextMenuEnabled) &&
-         !params_.selection_text.empty();
+         ai_chat::AIChatTabHelper::FromWebContents(source_web_contents_) &&
```

Comment:
For now (but not forever), if we're not creating a Tab Helper for this WebContents, then we're not enabling Leo in this context, so don't show those items. It's guarded on the same `IsRegularProfile` and `ai_chat::IsAIChatEnabled` that this function is already checking, but seems like we don't need to duplicate that check at the moment.